First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request consists of the first commits to create this repository from the skeleton.

## 💭 Motivation and context ##

See cisagov/cool-system#160.

## 🧪 Testing ##

I was able to successfully create the tunnel resources in our staging COOL environment.  The tunnel is not yet functional, since that requires:
* IPv4 space deconfliction
* VENOM folks to stand up their end of the tunnel
* VENOM folks to get back to me regarding which side initiates the tunnel

I also need to determine if the tunnel will automatically create the necessary route and propagate it to the default route table, or if I need to manually create the route.  That will fall out when we actually test with the VENOM folks.

This is a good start, though, and we can create issues and PRs for necessary changes as we go forward.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

jsf9k committed:
Rename repository references after clone.

jsf9k committed:
Modify TF code to create the s2s VPN tunnel to the VENOM environment

jsf9k committed:
Update README for this repo

Also remove an unnecessary variable and fix a typo in a variable
description.

jsf9k committed:
Reduce codeowners to just the folks familiar with this repo

dav3r commented on README.md:
```diff
@@ -47,39 +35,29 @@ module "example" {
-| aws_region | The AWS region to deploy into (e.g. us-east-1) | `string` | `us-east-1` | no |
-| subnet_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0) | `string` | n/a | yes |
-| tags | Tags to apply to all AWS resources created | `map(string)` | `{}` | no |
+| aws_region | The AWS region where the shared services account is to be created (e.g. "us-east-1"). | `string` | `us-east-1` | no |
```
I don't love the way this reads (`where the shared services account is to be created`).  Aren't we writing this as if `cool-sharedservices-networking` is already deployed?  How about: `The AWS region where Shared Services account resources reside`?

jsf9k replied:
See commit 3d15ed6.

dav3r replied:
That works- thanks! 👍

dav3r commented on README.md:
```diff
@@ -1,39 +1,27 @@
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This is a Terraform deployment for creating the site-to-site VPN
+tunnel between the COOL and the VENOM environment in the COOL Shared
```
```suggestion
tunnel between the COOL and the VENOM (Virtual Enterprise
Network Operations Manager) environment in the COOL Shared
```

dav3r commented on remote_states.tf:
```diff
@@ -0,0 +1,35 @@
+    key            = "cool-sharedservices-networking/terraform.tfstate"
+  }
+
+  workspace = terraform.workspace
```
In the past, we have preferred to rely on the name of the account to determine the workspace type (whenever possible), rather than on `terraform.workspace` - see https://github.com/cisagov/cool-pca-transitgateway-attachment/blob/3c78cf95de7383482a893105f620c86c734f7c37/remote_states.tf#L22-L72

mcdonnnj replied:
https://github.com/cisagov/cool-pca-transitgateway-attachment/blob/3c78cf95de7383482a893105f620c86c734f7c37/locals.tf#L46 for that last piece.

jsf9k replied:
See commit a2ec66e.

dav3r commented on s2s_vpn_tunnel.tf:
```diff
@@ -0,0 +1,59 @@
+  provider = aws.sharedservicesprovisionaccount
+
+  bgp_asn    = 65000 # Unused
+  ip_address = "52.177.137.15"
```
Should this be a Terraform variable?

jsf9k replied:
Well this code is VENOM-specific, and that IP address is _the_ IP for VENOM's end of the tunnel.  If I were going to make it a variable then I should make all the other tunnel parameters variables as well, although I don't think that makes this code any more reusable.

dav3r replied:
I was thinking that the VENOM folks would probably prefer that their IP address was not included in our code.  Also, if it ever changes, it would not require a code change on our end (kind of a weak argument I know, but it's true).

I understand that this isn't consistent with the other tunnel parameter variables (which I don't think need to be TF variables), but IP addresses can fall into the category of "sensitive data".  Not saying we are hiding it, but no need to advertise it more than necessary.

Anyway, I'm not going to die on this hill if you feel strongly about this.

jsf9k replied:
See commit 4be2fe0.

dav3r reviewed (approved):
Looks great!  🐍 

I left a few comments for your consideration.

jsf9k committed:
Define VENOM acronym

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Tweak variable description

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj commented on README.md:
```diff
@@ -47,39 +36,29 @@ module "example" {
-| tags | Tags to apply to all AWS resources created | `map(string)` | `{}` | no |
+| aws_region | The AWS region where the shared services account is to be created (e.g. "us-east-1"). | `string` | `us-east-1` | no |
+| tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
+| venom_vpn_preshared_key | The pre-shared key to use for setting up the site-to-site VPN connection between the COOL and VENOM.  This must be a string of 36 characters, which can include alphanumerics, periods, and underscores (e.g. "abcdefghijklmnopqrstuvwxyz0123456789"). | `string` | n/a | yes |
```
If we're giving an example, then I think we should include everything we said it can contain. Optionally we could shorten it with regex style characters classes to "[a-z][0-9]._", but that's just a thought.

```suggestion
| venom_vpn_preshared_key | The pre-shared key to use for setting up the site-to-site VPN connection between the COOL and VENOM.  This must be a string of 36 characters, which can include alphanumerics, periods, and underscores (e.g. "abcdefghijklmnopqrstuvwxyz0123456789._"). | `string` | n/a | yes |
```

jsf9k replied:
See commit 4be2fe0.

mcdonnnj reviewed (approved):
Looking good!

jsf9k committed:
Add a remote backend

I can't believe I failed to do this before.  Doh!

jsf9k committed:
Make the IP address of the VENOM tunnel endpoint a variable

Also add the period and underscore characters to the
venom_vpn_preshared_key variable example.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Compute the workspace type (prod or staging) based on the account name

Then use local.workspace_type instead of terraform.workspace in
remote_states.tf wherever possible.

Co-authored-by: Dave Redmin <david.redmin@trio.dhs.gov>
Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

dav3r commented on locals.tf:
```diff
@@ -0,0 +1,50 @@
+}
+
+# ------------------------------------------------------------------------------
+# Retrieve the information for all accouts in the organization.  This
```
Fix a typo:
```suggestion
# Retrieve the information for all accounts in the organization.  This
```

dav3r reviewed (approved):
Everything looks great now, aside from that typo I mentioned.  Excellent work! ✌️

jsf9k committed:
Fix a typo

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj commented on README.md:
```diff
@@ -47,39 +36,30 @@ module "example" {
+| aws_region | The AWS region where the Shared Services account resides (e.g. "us-east-1"). | `string` | `us-east-1` | no |
+| tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
+| venom_tunnel_ip | The IP address of the site-to-site VPN tunnel endpoint on the VENOM side (e.g. "100.200.75.25") | `string` | n/a | yes |
+| venom_vpn_preshared_key | The pre-shared key to use for setting up the site-to-site VPN connection between the COOL and VENOM.  This must be a string of 36 characters, which can include alphanumerics, periods, and underscores (e.g. "abcdefghijklmnopqrstuvwxyz0123456789"). | `string` | n/a | yes |
```
I guess when you force pushed this got lost in the mix?

```suggestion
| venom_vpn_preshared_key | The pre-shared key to use for setting up the site-to-site VPN connection between the COOL and VENOM.  This must be a string of 36 characters, which can include alphanumerics, periods, and underscores (e.g. "abcdefghijklmnopqrstuvwxyz0123456789._"). | `string` | n/a | yes |
```

jsf9k replied:
I updated the variable but not the README.  See commit 8d6c391, and note that the example PSK must be 36 characters in length.  (At least that's how I interpret VENOM's documentation.)

mcdonnnj replied:
Did you want to change the example to conform to the length requirement? I just wanted to make sure every class of input is shown.

jsf9k replied:
I _did_ change the example to conform with the length requirement.  See commits 4be2fe0 and 8d6c391.

mcdonnnj replied:
Ah, whoops. I misunderstood what was going on.

mcdonnnj reviewed (approved):
Looking even better! ✔

jsf9k committed:
Update README to match description of PSK variable

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>
